src/backend/InvenTree/order/models.py

```diff
@@ -599,13 +630,181 @@ def add_line_item(
+            extra_info = ''
+            if self.project_code:
+                extra_info = _(f'Please add a responsible to {self.project_code} or ')
+            extra_info += _('set a master approver group in settings.')
+
+            raise ValidationError(_(f'No eligible approvers. {extra_info}'))
+        else:
+            self.status = PurchaseOrderStatus.IN_APPROVAL.value
+            self.save()
+
+            trigger_event('purchaseorder.request_approval')
+            notify_users(
+                approver or [master_users],
+                self,
+                PurchaseOrder,
+                content=InvenTreeNotificationBodies.PurchaseOrderApprovalRequested,
+                delta=timedelta(seconds=60),
+                exclude=user,
+            )
+
+    def _action_approve(self, *args, **kwargs):
+        """Marks the PurchaseOrder as READY.
+
+        Order must currently be IN_APPROVAL
+        """
+        master_approvers = getSetting('PURCHASE_ORDER_APPROVE_ALL_GROUP')
+
+        if not self.requires_approval:
+            raise ValidationError(_('Approvals are not active'))
+        if not self.is_pending_approval:
+            raise ValidationError(_('Order needs to be in "Pending approval" state'))
+
+        if not self.approved_by:
+            permitted = False
+            approver = kwargs.pop('user', None)
+
+            if approver.groups.filter(name=master_approvers).exists():
+                permitted = True
+            elif hasattr(self.project_code, 'responsible'):
+                if self.project_code.responsible.is_user_allowed(approver):
+                    permitted = True
+
+            if permitted:
+                self.approved_by = approver
+                self.approved_date = InvenTree.helpers.current_date()
+                self.reject_reason = ''
+                self._action_ready()
+                trigger_event('purchaseorder.approved')
+                notify_users(
+                    [self.created_by, self.responsible],
+                    self,
+                    PurchaseOrder,
+                    InvenTreeNotificationBodies.PurchaseOrderApproved,
+                    exclude=approver,
+                    delta=timedelta(seconds=60),
+                )
+            else:
+                raise PermissionDenied()
+
+        else:
+            self._action_ready()
+
+    def _action_ready(self, *args, **kwargs):
+        """Marks the PurchaseOrder as READY.
+
+        Order must currently be PENDING or IN_APPROVAL
+        """
+        if not self.requires_approval and not self.can_be_ready:
+            raise ValidationError(_('Ready state setting not active'))
+
+        if self.is_pending or (self.is_pending_approval and self.approved_by):
+            self.status = PurchaseOrderStatus.READY.value
+            self.save()
+
+            user = kwargs.pop('user', None)
+
+            purchaser_group_name = getSetting('PURCHASE_ORDER_PURCHASER_GROUP')
```

Comment:
Once you have merged master in please use the new `common.settings.get_global_setting` method here - `getSetting` has been deprecated

Reply:
Will do

Reply:
@LavissaWoW for reference this will required https://github.com/inventree/InvenTree/pull/7404 to be merged in first